### ISSUE TITLE
Blazor Wasm App publish is failing with '--self-contained false --runtime linux-arm' on missing DotNetJsVersion

### ISSUE THREAD

**Issue by Reveon** [dotnet/aspnetcore#39284]
### Describe the bug
dotnet publish of a new Blazor Wasm app with parameters `--self-contained false --runtime linux-arm` is failing because of missing **DotNetJsVersion** in **ComputeBlazorPublishAssets** task

The only reference that I could find for DotNetJsVersion is in this PR dotnet/sdk#21919 so maybe there is some connection to the error I am getting

I am using the latest version of Visual Studio (17.0.4) and .NET (6.0.101)

### To Reproduce
1) Create new Blazor Wasm project in Visual Studio with default settings

![obrazek](https://user-images.githubusercontent.com/29383312/147675438-935b48c9-adb3-4302-8f92-1a993c9d9048.png)

2) Add `<PublishTrimmed>false</PublishTrimmed>` to csproj to avoid error NETSDK1102
3) run ` dotnet publish --self-contained false --configuration debug --runtime linux-arm`

### Exceptions (if any)
```
C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(412,5): error MSB4044: The "ComputeBlazorPublishAssets" task was not given a value for the required parameter "DotNetJsVersion". [C:\Source\Projects\BlazorPublishIssue\BlazorPublishIssue.csproj]
```

### Further technical details
Visual Studio 17.0.4

**dotnet --info**
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.101\

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  3.1.415 [C:\Program Files\dotnet\sdk]
  5.0.401 [C:\Program Files\dotnet\sdk]
  5.0.403 [C:\Program Files\dotnet\sdk]
  6.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by pranavkm** [comment]
BlazorWASM apps target the `browser-wasm` runtime and are always self-contained (the browser does not carry a runtime). Targeting a linux platform doesn't really make sense for it either.

**Comment by Reveon** [comment]
@pranavkm I get the same error also for the server project (when I create Asp.Net hosted Blazor app)

![obrazek](https://user-images.githubusercontent.com/29383312/147982866-eea170bd-d9ca-4611-b130-6b358b382bdf.png)

**Comment by pranavkm** [comment]
I guess the Blazor SDK has to mark `SelfContained` as a private property. You have a couple of options to resolve this:

a) You can avoid specifying `--self-contained false` as a publish option since it's false by default for the hosted app.
b) You could modify the wasm app to ignore the value of SelfContained specified from the CLI:

```diff
- <Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+ <Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly" TreatAsLocalProperty="SelfContained">

  <PropertyGroup>
+    <SelfContained>true</SelfContained>
  </PropertyGroup>
```

**Comment by Reveon** [comment]
the first one gives me this warning (but it builds anyway) 
` warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used.`

but the other one works perfectly (also the publish size is much smaller then for the first option). Thank you!

**Comment by silentdevnull** [comment]
Good Morning,

Does anyone have this working? I can publish to folder in Rider and everything works fine. If I do it from a build script on linux I receive the same error about DotNetJsVersion. 

When I publish to folder in Rider I have self contained set to true.

Any help is appreciated!!!!

Thank you

**Comment by KhoaDQ** [comment]
> I guess the Blazor SDK has to mark `SelfContained` as a private property. You have a couple of options to resolve this:
> 
> a) You can avoid specifying `--self-contained false` as a publish option since it's false by default for the hosted app. b) You could modify the wasm app to ignore the value of SelfContained specified from the CLI:
> 
> ```diff
> - <Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
> + <Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly" TreatAsLocalProperty="SelfContained">
> 
>   <PropertyGroup>
> +    <SelfContained>true</SelfContained>
>   </PropertyGroup>
> ```

I solved it, thanks

**Comment by mkArtakMSFT** [comment]
We believe this is addressed already in 8.0.
